airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/tools.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import os
+import re
+from typing import Any, Mapping, Optional, Union
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum as pdm
+
+from .exceptions import ShopifyBulkExceptions
+
+# default end line tag
+END_OF_FILE: str = "<end_of_file>"
+BULK_PARENT_KEY: str = "__parentId"
+
+
+class BulkTools:
+    @staticmethod
+    def camel_to_snake(camel_case: str):
+        snake_case = []
+        for char in camel_case:
+            if char.isupper():
+                snake_case.append("_" + char.lower())
+            else:
+                snake_case.append(char)
+        return "".join(snake_case).lstrip("_")
+
+    @staticmethod
+    def filename_from_url(job_result_url: str) -> str:
+        # Regular expression pattern to extract the filename
+        filename_pattern = r'filename\*?=(?:UTF-8\'\')?"([^"]+)"'
+        parsed_url = dict(parse_qsl(urlparse(job_result_url).query))
+        match = re.search(filename_pattern, parsed_url.get("response-content-disposition")).group(1)
+        if match:
+            return match
+        else:
+            raise ShopifyBulkExceptions.BulkJobResultUrlError(
+                f"Could not extract the `filename` from `result_url` provided, details: {job_result_url}",
+            )
+
+    @staticmethod
+    def from_iso8601_to_rfc3339(record: Mapping[str, Any], field: str) -> Mapping[str, Any]:
+        """
+        Converts date-time as follows:
+            Input: "2023-01-01T15:00:00Z"
+            Output: "2023-01-01T15:00:00+00:00"
+        If the value of the `field` is `None` we return it `as is`.
+        """
+        target_value = record.get(field)
+        return pdm.parse(target_value).to_rfc3339_string() if target_value else record.get(field)
```

Comment:
Added raising `Exception` if `TypError` or `ValueError` are hit.

Reply:
Actually, returned to the previous state, since we have to deal with the nested objects while processing records, and return `None` to the fields, that are missing in records, but have to be present by the schema.